### PR TITLE
readme: Various improvements

### PR DESCRIPTION
**Pull request by phlogistonjohn**
I've tried to spruce up the README.md file a bit. The major changes are a new, but short, introductory section and I've moved some of the "documentation" to a secondary page as they were more hints/tips-n-tricks level stuff than complete docs. Commit messages all have more detail.

### CODE REVIEW

**Review by ansiwen — APPROVED** [review]
LGTM

**Review by agarwal-mudit — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ To quickly test if one can build with go-ceph on your system, run:
 go get github.com/ceph/go-ceph
 ```
 
+Once compiled, code using go-ceph is expected to dynamically link to the Ceph
+libraries. These libraries must be available on the system where the go based
+binaries will be run. The use of cgo does not allow for fully static binaries.
```
Comment (nixpanic) [inline README.md:43]: I think this is not completely correct. It is probably possible to link C libraries statically into the GO executable. However, Ceph (packages) does not provide the option to statically link the libraries (no `.a` files, only `.so` ones).

**Review by nixpanic — APPROVED** [review]
Added a nit which is more a FYI, nothing that requires correcting.

File: README.md (hunk 1 above)
Comment (phlogistonjohn) [reply README.md:43]: I will change it to "Our use of cgo and ceph libraries" instead of "The use". I think that makes it more accurate.

Comment (phlogistonjohn) [reply README.md]: Updated. If you have issues with the new version please give a shout. Otherwise, I'll merge later this week.

**Pull request merged**